src/snowflake/snowpark/_internal/data_source_utils.py

```diff
@@ -313,7 +313,7 @@ def data_source_data_to_pandas_df(
     def df_map_method(pandas_df):
         return (
             pandas_df.applymap
-            if version.parse(pd.__version__) < version.parse("2.1.0")
+            if get_sorted_key_for_version(str(pd.__version__)) < (2, 1, 0)
```

Comment:
logged a jira, SNOW-1963784
once we merge the two connectors, we should use packaging.version.parse which is more reliable